article/app/model/dotcomponents/DotcomponentsDataModel.scala

```diff
@@ -449,6 +464,15 @@ object DotcomponentsDataModel {
       allTags
     )
 
+    val commercialConfig = CommercialConfiguration(
```

Comment:
I'm trying to think if there's a way to avoid needing to pass both a context and a whole javacriptpage to the datamodel constructor 🤔 it's quite a lot of stuff to give the model that it doesn't really care about. And 'context' has a habit of infecting the codebase.

Is there any way we could refactor this so that we build the commercial config JavaScript page *before* we try to create the datamodel, and then pass those values into the datamodel constructor? This way we could get rid of having to give the datamodel a context and javascriptPage, and instead just give it only what it needs, the commercial value?

Although, maybe this wouldn't be a good idea because it'd mean we're treating commercial config in a special way.

Maybe @nicl might have an opinion? I'm just wary of passing such generic constructs like context right down to the datamodel. The code is getting pretty convoluted in there!

Reply:
Good points. Will refactor in the next commit.